### PR TITLE
V2: Bug fix for hyperparameter optimization

### PR DESCRIPTION
**Pull request by hwpang**
## Description
@shihchengli identified some bugs while working on HPO benchmarking. This PR intends to address these bugs.

## Bugfix / Desired workflow
- The name for hyperparameter optimization should be `ffn_hidden_dim` and `hidden_dim`, not `ffn_hidden_size` and `hidden_size`. The latter were used in v1, and v2 uses different names.
- The search space for `ffn_num_layers` should be defined with `quniform`, not `randint`, to be consistent with v1.

## Questions
n/a

## Relevant issues
https://github.com/chemprop/chemprop/issues/791

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by shihchengli** [comment]
1. The `hidden_dim` should be `message_hidden_dim`
2. I think we should raise an error if incorrect parameters have been used(probably related to the `update_args_with_config()`

**Comment by shihchengli** [comment]
Should we also add early stopping and gradient clip in HPO? They are used in train.py (see [here](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/cli/train.py#L813-L814)), but not used in hpopt.py.

File: chemprop/cli/hpopt.py
Comment (shihchengli) [inline chemprop/cli/hpopt.py:113]:
```suggestion
        basic - the default set of hyperparameters for search: depth, ffn_num_layers, dropout, message_hidden_dim, and ffn_hidden_dim.
```

**Comment by hwpang** [comment]
> Should we also add early stopping and gradient clip in HPO? They are used in train.py (see [here](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/cli/train.py#L813-L814)), but not used in hpopt.py.

We have gradient clip in HPO: https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/cli/hpopt.py#L246.

**Comment by shihchengli** [comment]
Do you think we need to change the default value for the grace period?

**Comment by hwpang** [comment]
@shihchengli Good call! we can change these values for sure. I see in your issue that you suggest changing the grace period to 5 or 10. We can also discuss this in person / in a dev meeting

**Comment by KnathanM** [comment]
I was able to run `chemprop hpopt -i data.csv --splits-file splits.json --epochs 50 --hpopt-save-dir hyperparam_search_test --raytune-num-samples 30 --raytune-search-algorithm hyperopt --raytune-use-gpu --search-parameter-keywords all` on the logP dataset after checking out this branch, so I would approve once the tests pass.

**Review by shihchengli — APPROVED** [review]

**Comment by shihchengli** [comment]
I think they are the same. I don't have a strong preference for using either one. @hwpang, perhaps you can revert to using `randint`, adjusting the upper value by adding 1, so that it aligns with the style used for `depth`.

**Comment by hwpang** [comment]
I prefer to change depth to use `quniform`. This is what we had in v1: `"depth": hp.quniform("depth", low=2, high=6, q=1),`, and I accidentally use `randint` when I implement in v2. Additionally, I think it's a lot more intuitive to have an inclusive upper bound.

**Comment by davidegraff** [comment]
> I prefer to change depth to use `quniform`. This is what we had in v1: `"depth": hp.quniform("depth", low=2, high=6, q=1),`, and I accidentally use `randint` when I implement in v2. Additionally, I think it's a lot more intuitive to have an inclusive upper bound.

If that's your preference, fair enough, I would just ask that you make the style consistent throughout the code

**Comment by shihchengli** [comment]
Then, the search range for `depth` now ranges from 2 to 6. Is that correct?

**Comment by hwpang** [comment]
@shihchengli Yes, this is correct and is consistent with Chas's PR here: https://github.com/chemprop/chemprop/pull/571/files#diff-256076e9b835920c3b444bb87e0b7547a77118d45896dc74f104efe743184b91L31.

**Comment by davidegraff** [comment]
re: incredibly deep search network search space. I still have yet to receive a reply to [this comment](https://github.com/chemprop/chemprop/pull/734#discussion_r1558081394), both on #571 and #734

**Comment by hwpang** [comment]
@davidegraff Thanks for the reminder! Yeah, I agree that the search space is too large. I have since changed the search space for ffn num layers back to the v1 default, which is `"ffn_num_layers": tune.quniform(lower=1, upper=3, q=1),`.

**Pull request merged**
